lib/rubocop/cop/rspec/expectation_target_method.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks ExpectationTarget method.
+      #
+      # @example
+      #   # bad
+      #   expect(something).kind_of? Foo
+      #   is_expected == 42
+      #
+      #   # good
+      #   expect(something).to be_a Foo
+      #   is_expected.to eq 42
+      #
+      class ExpectationTargetMethod < Base
+        MSG = 'Do not use any method other than `.to` or `.not_to`.'
+        RESTRICT_ON_SEND = %i[expect is_expected].freeze
+
+        # @!method expect?(node)
+        def_node_matcher :expect?, <<~PATTERN
+          {
+            (send nil? :expect ...)
+            (send nil? :is_expected)
+          }
+        PATTERN
+
+        # @!method expect_block?(node)
+        def_node_matcher :expect_block?, <<~PATTERN
+          (block #expect? (args) _body)
+        PATTERN
+
+        # @!method expectation_target_method?(node)
+        def_node_matcher :expectation_target_method?, <<~PATTERN
```

Comment:
```suggestion
        def_node_matcher :expectation_without_runner?, <<~PATTERN
          (send #expect? !#Runners.all …)
```

Should be able to detect expects what its name tells.

Reply:
Thank you for the suggestion. That looks better, so I will fix it.